src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,49 @@
+package seedu.addressbook.data.person;
+
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+    protected boolean isPrivate;
+
+    public final String value;
+
+    /**
+     * Validates given property.
+     *
+     * @throws IllegalValueException if given email address string is invalid.
+     */
+    public Contact(String property, boolean isPrivate, String MESSAGE_PROPERTY_CONSTRAINTS, String PROPERTY_VALIDATION_REGEX) throws IllegalValueException {
+        this.isPrivate = isPrivate;
+        String trimmedProperty = property.trim();
+        if (!isValidProperty(trimmedProperty, PROPERTY_VALIDATION_REGEX)) {
+            throw new IllegalValueException(MESSAGE_PROPERTY_CONSTRAINTS);
+        }
+        this.value = trimmedProperty;
+    }
+
+    /**
+     * Returns true if the given string is a valid person property.
+     */
+    public static boolean isValidProperty(String test, String PROPERTY_VALIDATION_REGEX) {
+        return test.matches(PROPERTY_VALIDATION_REGEX);
+    }
+
+    public String toString() {
+        return value;
+    }
+
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Phone // instanceof handles nulls
+                && this.value.equals(((Phone) other).value)); // state check
```

Comment:
Since Contact is the parent class, its not good OOPs paradigm to do an equals comparison based on Phone which is the child class. Think about how you can abstract the common operation to the superclass without it being specific to any one child class.